src/screens/NewTopicScreen.tsx

```diff
@@ -0,0 +1,156 @@
+import * as React from 'react';
+import { postNewTopic}from '../api/DataHandler';
+
+//import { Link } from 'react-router-dom'
+
+interface NewTopicState{
+    post_title: string,
+    post_text: string,
+    post_url: string,
+    username: string,
+    pwdHash: string,
+    hanesst_id: number,
+}
+class NewTopicScreen extends React.Component<any, NewTopicState> {
+
+state = {
+    post_title: "",
+    post_text: "",
+    post_url: "",
+    username: "",
+    pwdHash: "",
+    hanesst_id: 0
+}
+
+    
+handlePostTitleInput(event:React.FormEvent<HTMLInputElement>){
+    this.setState({post_title: event.currentTarget.value})
+}
+
+handlePostTextInput(event:React.FormEvent<HTMLInputElement>){
+    this.setState({post_text: event.currentTarget.value})
+}
+
+handlePostUrlInput(event:React.FormEvent<HTMLInputElement>){
+    this.setState({post_url: event.currentTarget.value})
+}
+
+handleUserNameInput(event:React.FormEvent<HTMLInputElement>){
+    this.setState({username: event.currentTarget.value})
+}
+
+handlePWDInput(event:React.FormEvent<HTMLInputElement>){
+    this.setState({pwdHash: event.currentTarget.value})
+}
+
+handleHasesstIdInput(event:React.FormEvent<HTMLInputElement>){
+    this.setState({hanesst_id: event.currentTarget.valueAsNumber})
+}
+
+
+  
+handleNewTopicSubmit = async ( e: React.FormEvent<HTMLFormElement>):Promise<void> => {
+        e.preventDefault();
+        if(this.state.hanesst_id <= 0  ){
+            alert("hanesst id can't be equal to or below 0!")
+        } else if(!this.state.username || this.state.username.length <= 3){
+            alert("Username must be at least 3 chars long")
+        } else if(!this.state.pwdHash || this.state.pwdHash.length <= 3){
+            alert("Password must be at least 3 chars long")
+        } 
+        else {
+            postNewTopic(this.state.username,this.state.pwdHash,this.state.post_title, this.state.post_url, this.state.post_text, this.state.hanesst_id).then(res => {
+                console.log(res);
+            });
+            alert("Your story was successfully posted!");
+        }
+    }
+
+    newTopic(){
+        return(
+            <div>
+
+                <h1>Welcome to the new Topic Screen!</h1>
+
+                                
+                <form  onSubmit={this.handleNewTopicSubmit}>
```

Comment:
Why are we using a form? We can integrate an onClick on a button, instead of enclosing the html in a form - and then just send the data with this.state.

Reply:
Old school thinking, if time should be changed, but it works for now.